src/non_blocking/gzip.rs

```diff
@@ -169,10 +157,7 @@ mod tests {
     use super::*;
     use crate::gzip::Encoder;
     use crate::util::{nb_read_to_end, WouldBlockReader};
-    #[cfg(feature = "no_std")]
     use core2::io::Write;
```

Comment:
```suggestion
    use core2::io::Write;
    #[cfg(not(feature = "std"))]
    use alloc::vec::Vec;
```

Reply:
Yes. This is not resolved. You still need the `use alloc::vec::Vec` import, even if you remove the conditional compilation.